File: day1/solution_p2.hs
main' = do
  xs <- map (read :: String -> Int) . lines <$> readFile "input"
  let ys = zipWith3 (\a b c -> a + b + c) xs (tail xs) (tail (tail xs))
  print . length $ filter (True ==) $ zipWith (<) ys $ tail ys

File: day1/solution_p2.rs
fn main() {
    println!("{}", depth_increase_count());
}

fn depth_increase_count() -> usize {
    std::fs::read_to_string("./input")
        .unwrap()
        .lines()
        .map(|num| num.parse::<u16>().unwrap())
        .collect::<Vec<_>>()
        .windows(3)
        .map(|w| w.iter().sum::<u16>())
        .collect::<Vec<_>>()
        .windows(2)
        .filter(|&op| op[0] < op[1])
        .count()
}
